### DISCUSSION TITLE
How do the translation bangs work?

### DISCUSSION THREAD

**Discussion by ranidspace** [searxng/searxng#3527] [Q&A] [answered]
Basically nothing I tried works at all for them they're the first things that show up in the "Engines" settings menu and I cannot figure it out at all. I saw the "currency" option there and I have no idea how that one works either but the translation one I want to use the most.

**Comment by Daniel-MFG** [comment]
Same question here...
I wanted to check the syntax in the docs (https://docs.searxng.org/user/search-syntax.html) but it seems like they aren't added there yet...

So currently all i get if i use "!dc" or "!translate" with a word is:
`Sorry!

No results were found. You can try to:

Refresh the page.
Search for another query or select another category (above).
Change the search engine used in the preferences: [/preferences](https://bakasearch.ddns.net/preferences)
Switch to another instance: https://searx.space/`

**Comment by return42** [comment, marked as answer]
Apparently we are still missing a documentation for the [Online Dictionaries](https://docs.searxng.org/dev/engines/index.html#online-dictionary)

Online Dictionaries need a `<from>-<to>` in the search query:

https://github.com/searxng/searxng/blob/837f3bcd101274b7cbea95e6ab2c19282e8b2937/searx/search/processors/online_dictionary.py#L11

By example: `en-de The quick brown fox jumps over the lazy dog` will use all online dictionaries activated by default.

Like in normal search queries you can select explicit an engine: `!lv en-de The quick brown fox jumps over the lazy dog`

To activate all online dictionaries in the group [translate](https://docs.searxng.org/user/configured_engines.html#group-translate) use  `!translate en-de hello`
